problem_builder/step.py

```diff
@@ -104,13 +113,42 @@ def allowed_nested_blocks(self):
             AnswerRecapBlock, MentoringTableBlock,
         ]
 
-    @property
-    def steps(self):
-        """ Get the usage_ids of all of this XBlock's children that are "Questions" """
-        from mixins import QuestionMixin
-        return [
-            _normalize_id(child_id) for child_id in self.children if child_isinstance(self, child_id, QuestionMixin)
-        ]
+    @XBlock.json_handler
+    def submit(self, submissions, suffix=''):
+        log.info(u'Received submissions: {}'.format(submissions))
+
+        # Submit child blocks (questions) and gather results
+        submit_results = []
+        for child in self.get_steps():
+            if child.name and child.name in submissions:
+                submission = submissions[child.name]
+                child_result = child.submit(submission)
+                submit_results.append([child.name, child_result])
+                child.save()
+
+        # Update results stored for this step
+        while self.student_results:
+            self.student_results.pop()
+        for result in submit_results:
+            self.student_results.append(result)
+
+        # Compute "answer status" for this step
+        if all(result[1]['status'] == 'correct' for result in submit_results):
+            completed = 'correct'
+        elif all(result[1]['status'] == 'incorrect' for result in submit_results):
+            completed = 'incorrect'
+        else:
+            completed = 'partial'
```

Comment:
@itsjeyd nit: slightly more concise:

    correct_count = sum(1 for result in submit_results if result[1]['status'] == 'correct' )
    if correct_count == len(self.questions): completed = 'correct'
    elif correct_count > 0: completed = 'partial'
    else: completed = 'incorrect'

Still three-branch if, but at least it goes over results only once. Those are a dict, so performance benefits should be negligible. Feel free to ignore this